core/input.js

```diff
@@ -118,6 +118,7 @@ Blockly.Input.prototype.insertFieldAt = function(index, field, opt_name) {
   if (this.sourceBlock_.rendered) {
     field.init();
   }
+  field.initValue();
```

Comment:
Just a heads up for you: I'm was thinking about changing this area a little bit (even before you put up this PR). It works perfectly for now, I just wanted to warn you so I wasn't stepping on toes.
And I'll put up an issue before I do anything so that we can discuss.